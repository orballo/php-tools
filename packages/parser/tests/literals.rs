// mod common;

// use common::Extractor;
// use insta::{assert_yaml_snapshot, glob, with_settings};
// use parser::{PHPParser, PHPRule};
// use pest::Parser;
// use std::fs;

// #[test]
// fn single_quoted_string_literal() {
//     glob!("fixtures/single_quoted_string_literals.php", |path| {
//         let file = fs::read_to_string(path).unwrap();
//         let pairs = PHPParser::parse(PHPRule::SCRIPT, &file)
//             .expect("Something went wrong parsing the file");
//         let literals = Extractor::default()
//             .extract_nodes(pairs, PHPRule::SINGLE_QUOTED_STRING_LITERAL);

//         with_settings!({prepend_module_to_snapshot => false}, {
//             assert_yaml_snapshot!(literals);
//         })
//     });
// }

// #[test]
// fn double_quoted_string_literal() {
//     glob!("fixtures/double_quoted_string_literals.php", |path| {
//         let file = fs::read_to_string(path).unwrap();
//         let pairs = PHPParser::parse(PHPRule::SCRIPT, &file)
//             .expect("Something went wrong parsing the file");
//         let literals = Extractor::default()
//             .extract_nodes(pairs, PHPRule::DOUBLE_QUOTED_STRING_LITERAL);

//         with_settings!({prepend_module_to_snapshot => false}, {
//             insta::assert_yaml_snapshot!(literals);
//         })
//     });
// }

// #[test]
// fn heredoc_string_literal() {
//     glob!("fixtures/heredoc_string_literals.php", |path| {
//         let file = fs::read_to_string(path).unwrap();
//         let pairs = PHPParser::parse(PHPRule::SCRIPT, &file)
//             .expect("Something went wrong parsing the file");
//         let literals = Extractor::default()
//             .extract_nodes(pairs, PHPRule::HEREDOC_STRING_LITERAL);

//         with_settings!({prepend_module_to_snapshot => false}, {
//             insta::assert_yaml_snapshot!(literals);
//         })
//     });
// }

// #[test]
// fn nowdoc_string_literal() {
//     glob!("fixtures/nowdoc_string_literals.php", |path| {
//         let file = fs::read_to_string(path).unwrap();
//         let pairs = PHPParser::parse(PHPRule::SCRIPT, &file)
//             .expect("Something went wrong parsing the file");
//         let literals =
//             Extractor::default().extract_nodes(pairs, PHPRule::STRING_LITERAL);

//         with_settings!({prepend_module_to_snapshot => false}, {
//             insta::assert_yaml_snapshot!(literals);
//         })
//     });
// }
