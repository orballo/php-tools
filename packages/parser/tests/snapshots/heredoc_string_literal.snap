---
source: packages/parser/tests/literals.rs
expression: literals
input_file: packages/parser/tests/fixtures/heredoc_string_literals.php
---
- name: Basic heredoc string with double quotes
  content: "<<<\"EOD\"\nThis is a basic heredoc string.\nIt can span multiple lines.\nEOD"
  line: 4
  length: 72
- name: Basic heredoc string without double quotes
  content: "<<<EOD\nThis is a basic heredoc string.\nIt can span multiple lines.\nEOD"
  line: 10
  length: 70
- name: ""
  content: "<<<EOD\nHello, $name!\nVariables are not interpolated in heredoc strings.\nEOD"
  line: 17
  length: 75
- name: Heredoc with quotes and special characters
  content: "<<<EOD\nHeredoc strings can contain \"double quotes\" and 'single quotes'.\nSpecial characters like \\n \\t are treated literally.\nEOD"
  line: 23
  length: 128
- name: Heredoc with HTML
  content: "<<<HTML_CODE\n<div class=\"example\">\n    <p>This is an HTML snippet in a heredoc string.</p>\n</div>\nHTML_CODE"
  line: 29
  length: 107
- name: Heredoc with indentation (preserves whitespace)
  content: "<<<PHP_CODE\n    function example() {\n        echo \"This code block preserves indentation\";\n    }\nPHP_CODE"
  line: 36
  length: 105
- name: Empty heredoc string
  content: "<<<EMPTY\nEMPTY"
  line: 43
  length: 14
- name: Heredoc containing PHP code
  content: "<<<PHP_SCRIPT\n<?php\n$array = [1, 2, 3];\nforeach ($array as $value) {\n    echo $value;\n}\n?>\nPHP_SCRIPT"
  line: 47
  length: 101
